sdk/core/Azure.Core/src/TokenRequestContext.cs

```diff
@@ -47,5 +61,10 @@ public TokenRequestContext(string[] scopes, string? parentRequestId = default, s
         /// Additional claims to be included in the token. See <see href="https://openid.net/specs/openid-connect-core-1_0-final.html#ClaimsParameter">https://openid.net/specs/openid-connect-core-1_0-final.html#ClaimsParameter</see> for more information on format and content.
         /// </summary>
         public string? Claims { get; }
+
+        /// <summary>
+        /// A hint to indicate which tenantId is preferred.
+        /// </summary>
+        public string? TenantIdHint { get; }
```

Comment:
Perhaps we could just call this `TenantId` rather `TenantIdHint` and error if a credential doesn't support multi-tenant auth and the value is different from the configured value.